exasol_sphinx_github_pages_generator/deployer.py

```diff
@@ -0,0 +1,192 @@
+
+class GithubPagesDeployer:
+    def __init__(self, source_dir, module_path, target_branch, push_origin, push_enabled, source_branch,
+                 current_commit_id, tempdir):
+        self.source_dir = source_dir
+        self.module_path = module_path
+        self.target_branch = target_branch
+        self.push_origin = push_origin
+        self.push_enabled = push_enabled
+        self.source_branch = source_branch
+        self.current_commit_id = current_commit_id.stdout
+        self.worktree_paths = {"target_worktree": tempdir + "/worktrees/worktree_target",
+                              "source_worktree": tempdir + "/worktrees/worktree_source"}
+        self.build_dir = tempdir + "/build"
+        self.intermediate_dir = tempdir + "/intermediate"
+        self.staged_changes = False
+
+    def check_out_source_branch_as_worktree(self,
+                                            source_branch_commit_id):
+        source_branch_exists_remote = run(["git", "show-branch", f"remotes/origin/{self.source_branch}"],
+                                          capture_output=True, text=True)
+        print("source_branch_exists_remote : " + str(source_branch_exists_remote))
+        if source_branch_exists_remote.returncode == 0:
+            try:
+                # "If <branch> does exist, it will be checked out in the new working tree, if it’s not checked out
+                # anywhere else, otherwise the command will refuse to create the working tree (unless --force is used)."
+                run(["git", "worktree", "add", self.worktree_paths["source_worktree"],
+                     self.source_branch, "--force"], check=True)
+                print(f"Successfully added new temp worktree for source branch {self.source_branch}, "
+                      f"and checked out (Local or stashed changes will be ignored in this build).")
+                os.chdir(f"{self.worktree_paths['source_worktree']}/doc")
+                current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True,
+                                     check=True)
+                print("new current_branch/detected source branch : " + str(current_branch.stdout))
+            except subprocess.CalledProcessError:
+                sys.exit("Problem with adding worktree for source")
+
+        elif source_branch_commit_id.returncode == 0:
+            sys.exit("Source branch exists locally, but not on remote, and source branch is not current branch."
+                     "Please push your source branch to remote.")
+        else:
+            sys.exit(f"source branch {self.source_branch} does not exist")
+
+    def detect_or_verify_source_branch(self):
+        current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True, check=True)
+        print(current_branch.stdout)
+        print(f"source branch before detect: {self.source_branch}")
+        if self.source_branch == "":
```

Comment:
if you formulate this abort conditions like this:
```python
if self.source_branch == "" and self.current_branch.stdout == """
   sys.exit(...)
```
You can save you indentation.

Btw if you check for empty strings you also can simplify the conditon to:

```python
if not self.source_branch or not self._current_branch.stdout:
   sys.exit(...)
```

Reply:
"Both true" and "first if true, second false" are two distinct cases though.

Reply:
did read into the entire logic, if that's the case sure then you still should nest. still often you can rewrite to pull out the `exit` conditions and write them at the beginning to prevent nesting further down.
Second suggestion which I should have done as a separate comment,

`if foo == ""` can be replaced with `if not foo`